src/ContainerTrait.php

```diff
@@ -14,22 +39,152 @@ public function _unique_element($desired = null)
         return $desired.($postfix > 1 ? ('_'.$postfix) : '');
     }
 
-    protected function _unique() {
+    /**
+     * Extension to add() method which will perform linking of
+     * the object with the current class.
+     */
+    protected function _add_Container($element, $args = [])
+    {
+        // Normalize the arguments, bring name out
+        if (is_string($args)) {
+
+            // passed as string
+            $args=[$args];
+        } elseif (!is_array($args)) {
+
+            throw new Exception(['Second argument must be array','arg2'=>$args]);
+        } elseif (is_array($args) && isset($args['name'])) {
```

Comment:
`is_array($args)` here is not needed anymore, because it'll always be array at this point ;)